src/test/java/core/basesyntax/service/impl/PurchaseFruitHandlerImplTest.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.dto.FruitRecordDto;
+import core.basesyntax.storage.DataBase;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class PurchaseFruitHandlerImplTest {
```

Comment:
better to add one more test when fruit is not yet in db to cover more cases

Reply:
Added